Sources/AppearanceManager.swift

```diff
@@ -81,4 +81,23 @@ extension UINavigationController {
     override open var childForStatusBarStyle: UIViewController? {
         return self.topViewController
     }
+
+    override open func viewDidLoad() {
+        super.viewDidLoad()
+        NotificationCenter.default.addObserver(self, selector: #selector(updateTheme), name: .VLCThemeDidChangeNotification, object: nil)
+        updateTheme()
+    }
+
+    @objc func updateTheme() {
+        if #available(iOS 11.0, *) {
+            navigationBar.largeTitleTextAttributes = [
+                NSAttributedString.Key.foregroundColor: PresentationTheme.current.colors.navigationbarTextColor
```

Comment:
without this code `navigationBar` appearance will be changed with lag (when switching to light, it will switch to dark, then to light when dark is expected) - very important line of code!